tritium-samples-tracing/src/main/java/com/palantir/tritium/samples/tracing/SamplesTracer.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.samples.tracing;
+
+import java.util.Optional;
+
+public class SamplesTracer {
+    public static Optional<String> getTraceIfObservable() {
+        if (!com.palantir.tracing.Tracer.isTraceObservable()) {
+            return Optional.empty();
+        }
+        return Optional.of(com.palantir.tracing.Tracer.getTraceId());
+    }
+}
```

Comment:
I would suggest avoiding the `Optional` allocation on hot paths here and just use empty string to indicate no trace ID and document the behavior.

```suggestion
import com.palantir.tracing.Tracer;

public final class SamplesTracer {
    /**
     * Returns the trace ID for this thread's trace if it is observable; otherwise a blank string.
     */
    public static String getTraceIfObservable() {
        if (Tracer.isTraceObservable()) {
            return Tracer.getTraceId();
        }
        return "";
    }

    private SamplesTracer() {}
}
```